lms/djangoapps/course_api/v0/views.py

```diff
@@ -0,0 +1,353 @@
+    """
+    **Use Case**
+
+        CourseDetail returns details for a course.
+
+        The optional **depth** parameter that allows clients to get child content down to the specified tree level.
+
+    **Example requests**:
+
+        GET /{course_id}/
+
+        GET /{course_id}/?depth=2
+
+    **Response Values**
+
+        * category: The type of content.
+
+        * name: The name of the course.
+
+        * uri: The URI to use to get details of the course.
+
+        * course: The course number.
+
+        * content: When the depth parameter is used, a collection of child
+          course content entities, such as chapters, sequentials, and
+          components.
+
+        * due:  The due date. For courses, the value is always null.
+
+        * org: The organization specified for the course.
+
+        * id: The unique identifier for the course.
+    """
+
+    serializer_class = CourseSerializer
+
+    def get_object(self, queryset=None):
+        course_id = self.kwargs.get('course_id')
+        request = self.request
+        course_descriptor = self.get_course_or_404(request, course_id)
+        depth = self.depth(request)
+        course_descriptor = self.fetch_children(course_descriptor, depth)
+        return course_descriptor
+
+
+class CourseGradedContent(CourseContentMixin, ApiKeyHeaderPermissionMixin, ListAPIView):
+    """
+    **Use Case**
+
+        Retrieves course graded content and filtered children.
+
+    **Example requests**:
+
+        GET /{course_id}/graded_content/?filter_children_category=problem
+
+    **Response Values**
+
+        * category: The type of content.
+
+        * name: The name of the content entity.
+
+        * uri: The URI of the content entity.
+
+        * id: The unique identifier for the course.
+
+        * children: Content entities that this content entity contains.
+
+        * format: The type of the content (e.g. Exam, Homework). Note: These values are course-dependent.
+          Do not make any assumptions based on assignment type.
+
+        * problems: {
+            * id: The ID of the problem.
+
+            * name: The name of the problem.
+        }
+    """
+
+    serializer_class = GradedContentSerializer
+    allow_empty = False
+
+    def _filter_children(self, node, **kwargs):
+        """ Retrieve the problems from the node/tree. """
+
+        matched = True
+        for name, value in kwargs.iteritems():
+            matched &= (getattr(node, name, None) == value)
+            if not matched:
+                break
+
+        if matched:
+            return [node]
+
+        if not node.has_children:
+            return []
+
+        problems = []
+        for child in node.get_children():
+            problems += self._filter_children(child, **kwargs)
+
+        return problems
+
+    def depth(self, request):
+        # Load the entire content tree since we will need to filter down to the leaf nodes.
+        return None
+
+    def get_queryset(self):
```

Comment:
@mulby @cpennington I have updated this view to be a bit more generic in regards to the children returned for each node. Instead of hard-coding the problems category, I have added a query string param to retrieve this value.

In answer to the obvious question of "why not just add filtering to the content endpoint," I still need the parent assignment, not just the child/problem node at the bottom of the tree.